### DISCUSSION TITLE
has anyone built a splittable tab application?

### DISCUSSION THREAD

**Discussion by shi-yan** [tauri-apps/tauri#6464] [Q&A] [answered]
I'm wondering if you have seen any splittable tab applications built using tauri? I want to know if it is possible to do this.

examples of splittable tab applications are chrome and sublime text where you can drag to split a tab into a new window. and use dropping to merge two separate windows into one.

I understand tauri supports multi-window apps. but I don't know if the drag-and-drop part is easy to implement.

**Comment by FabianLars** [comment, marked as answer]
Sooo, i can't give any examples here, just a heads-up: Tauri does not support multiple webviews inside the same window, so all that tabbing mechanism would be html/js only (which may make it easier for you to google stuff?)

**Reply by shi-yan** [reply]
Thank you. I looked at Tauri's drag and drop functions. While it can detect a drop event initiated from a different window, it doesn't seem to carry a payload. Also, it doesn't seem to give me the coordinate of where the drop event happened.

**Reply by FabianLars** [reply]
At least with the coordinates you should have more luck by disabling tauri's custom file drop by setting `"fileDropEnabled": false` and using the browser's native drag-drop apis. That likely won't help moving data from one tauri window from another but at least the coordinates should be right. Maybe you can use the drag/drop apis then to use tauri's [event system](https://tauri.app/v1/guides/features/events) to send the data around.

**Comment by pavi2410** [comment]
[Spacedrive](https://www.spacedrive.com/) has support for tabs. Looks like it's built using React in userland. https://github.com/spacedriveapp/spacedrive/blob/d79f5046d5a452f04975a38a7be9a7dd4a45ea2d/apps/desktop/src/App.tsx#L218

**Comment by lukigarazus** [comment]
@FabianLars and now that Tauri v2 has experimental support for multiple Webviews inside a single Window?

**Reply by xuchaoqian** [reply]
> @FabianLars and now that Tauri v2 has experimental support for multiple Webviews inside a single Window?

Yes, check this API: [add_child ](https://docs.rs/tauri/2.1.1/tauri/window/struct.Window.html#method.add_child)
